torch/_inductor/codegen/wrapper.py

```diff
@@ -2036,6 +2036,81 @@ def codegen_while_loop(self, while_loop):
         )
         self.writeline(ExitSubgraphLine(self))
 
+    def codegen_sequential_scan(self, sequential_scan):
+        from torch._higher_order_ops.scan import _extract_carry_and_out
+
+        name = sequential_scan.get_name()
+        outer_init = [buf.codegen_reference() for buf in sequential_scan.init]
+        outer_xs = [buf.codegen_reference() for buf in sequential_scan.xs]
+        outer_additional_inputs = [
+            buf.codegen_reference() for buf in sequential_scan.additional_inputs
+        ]
+        dim = sequential_scan.dim
+
+        self.writeline(f"{name} = [None] * {len(sequential_scan.outputs)}")
+        outer_outputs = [f"{name}[{i}]" for i in range(len(sequential_scan.outputs))]
+        outer_carry_out, outer_ys_out = _extract_carry_and_out(
+            outer_outputs, len(outer_init)
+        )
+        # Initialize the carry output to be a clone of init.
+        outer_carry_outs = []
+        inner_carry_outs = []
+        self.writeline(f"inner_carry_out = [None for i in range({len(outer_init)})]")
+        for i, init in enumerate(
+            _extract_carry_and_out(sequential_scan.outputs, len(outer_init))[0]
+        ):
+            self.writeline(f"inner_carry_out[{i}] = {outer_init[i]}.clone()")
```

Comment:
This clone should be handled during the lowering phase.

Reply:
Done!